bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -7512,6 +7551,35 @@ public void setImagePaths(long obj_id, List<String> paths) throws TskCoreExcepti
 		}
 	}
 
+	public void deleteDataSource(long obj_id) throws TskCoreException {
+        CaseDbConnection connection = connections.getConnection();
+		Statement statement = null;
+		Statement statement2 = null;
+		ResultSet resultSet = null;
+		acquireSingleUserCaseWriteLock();
+		try {
+			statement = connection.createStatement();
+			connection.beginTransaction();
+			statement.execute("DELETE FROM tsk_objects WHERE obj_id = " + Long.toString(obj_id));
+			statement.execute("DELETE FROM image_gallery_groups WHERE data_source_obj_id = " + Long.toString(obj_id));
+			String accountSQL = "SELECT account_id FROM accounts " +
+                                "WHERE account_id NOT IN (SELECT account1_id FROM account_relationships) " +
+                                "AND account_id NOT IN (SELECT account2_id FROM account_relationships);";
+            statement2 = connection.createStatement();
+			resultSet = connection.executeQuery(statement2, accountSQL); //NON-NLS
+			while (resultSet.next()) {
+				statement.execute("DELETE FROM accounts WHERE account_id = " + resultSet.getString("account_id"));
+			}
+			connection.commitTransaction();
```

Comment:
The reason this approach to deleting form the accounts table works needs to be documented with a comment.

Also, I don't like the idea of making a bunch of trips to the database in a while loop. I think that the loop can be eliminated with an SQL statement of the form "DELETE FROM accounts WHERE account_id IN (SELECT account_id FROM accounts...etc.)."

Reply:
Moved to one statement.

Reply:
> The reason this approach to deleting from the accounts table works needs to be documented with a comment.

What I'm driving at with this remark is that the "DELETE FROM accounts WHERE..." SQL is not intuitive. To save future maintainers the trouble of looking it up, we should explicitly state how the data source delete in tsk_objects cascades to deletion of rows in account_relationships.